### PR TITLE
kvclient/rangefeed: allow user to resume rangefeed with span frontier

### PR DESCRIPTION
**Pull request by msbutler**
This patch changes the semantics of the new StartFromFrontier rangefeed method to resume the rangefeed from the passed in frontier, after the initial scan completes. Previously the rangefeed would resume from the frontier's low water mark, which would result in wasted work from the previous rangefeed run.

Informs #124915

Release note: none

### CODE REVIEW

**Review by dt — APPROVED** [review]

**Review by stevendanna — APPROVED** [review]

**Comment by msbutler** [comment]
TFTR!

bors r=stevendanna

**Comment by msbutler** [comment]
Unrelated flake

bors r+

**Review by nvb — COMMENTED** [review]
LGTM

File: pkg/kv/kvclient/rangefeed/rangefeed.go [hunk 1]
```diff
@@ -291,7 +304,7 @@ func (f *RangeFeed) run(ctx context.Context, frontier span.Frontier) {
 		if failed := f.runInitialScan(ctx, &restartLogEvery, &r, frontier); failed {
 			return
 		}
-	} else {
+	} else if !resumeWithFrontier {
```
Comment (dt) [inline pkg/kv/kvclient/rangefeed/rangefeed.go:307]: Do we need this conditional here? what breaks if we unconditionally forward to the initial ts here?

Comment (msbutler) [reply pkg/kv/kvclient/rangefeed/rangefeed.go:307]: nothing breaks, just a tiny perf optimization that may or may not matter.

**Pull request merged**